pycgm/pycgm.py

```diff
@@ -0,0 +1,5263 @@
+
+        l_kneo, l_knex, l_kney, l_knez = knee_axis[4:]
+        axis_results[axis_idx["L KNEO"]] = l_kneo
+        axis_results[axis_idx["L KNEX"]] = l_knex
+        axis_results[axis_idx["L KNEY"]] = l_kney
+        axis_results[axis_idx["L KNEZ"]] = l_knez
+
+        # Populate RKnee and LKnee in joint_centers
+        joint_centers[jc_idx['RKnee']] = r_kneo
+        joint_centers[jc_idx['LKnee']] = l_kneo
+
+        rtib = frame[marker_idx[markers["RTIB"]]]
+        ltib = frame[marker_idx[markers["LTIB"]]]
+        rank = frame[marker_idx[markers["RANK"]]]
+        lank = frame[marker_idx[markers["LANK"]]]
+        knee_origin = np.array([knee_axis[0], knee_axis[4]])
+
+        ankle_axis = ank_ax(rtib, ltib, rank, lank, knee_origin, measurements)
+
+        # Populate left and right ankle axis results
+        r_anko, r_ankx, r_anky, r_ankz = ankle_axis[:4]
+        axis_results[axis_idx["R ANKO"]] = r_anko
+        axis_results[axis_idx["R ANKX"]] = r_ankx
+        axis_results[axis_idx["R ANKY"]] = r_anky
+        axis_results[axis_idx["R ANKZ"]] = r_ankz
+
+        l_anko, l_ankx, l_anky, l_ankz = ankle_axis[4:]
+        axis_results[axis_idx["L ANKO"]] = l_anko
+        axis_results[axis_idx["L ANKX"]] = l_ankx
+        axis_results[axis_idx["L ANKY"]] = l_anky
+        axis_results[axis_idx["L ANKZ"]] = l_ankz
+
+        # Populate RAnkle and LAnkle in joint_centers
+        joint_centers[jc_idx['RAnkle']] = r_anko
+        joint_centers[jc_idx['LAnkle']] = l_anko
+
+        rtoe = frame[marker_idx[markers["RTOE"]]]
+        ltoe = frame[marker_idx[markers["LTOE"]]]
+        rhee = frame[marker_idx[markers["RHEE"]]]
+        lhee = frame[marker_idx[markers["LHEE"]]]
+
+        # Populate RHEE and LHEE in joint_centers
+        joint_centers[jc_idx['RHEE']] = rhee
+        joint_centers[jc_idx['LHEE']] = lhee
+
+        foot_axis = foo_ax(rtoe, ltoe, ankle_axis, measurements)
+
+        # Populate left and right foot axis results
+        r_fooo, r_foox, r_fooy, r_fooz = foot_axis[:4]
+        axis_results[axis_idx["R FOOO"]] = r_fooo
+        axis_results[axis_idx["R FOOX"]] = r_foox
+        axis_results[axis_idx["R FOOY"]] = r_fooy
+        axis_results[axis_idx["R FOOZ"]] = r_fooz
+
+        l_fooo, l_foox, l_fooy, l_fooz = foot_axis[4:]
+        axis_results[axis_idx["L FOOO"]] = l_fooo
+        axis_results[axis_idx["L FOOX"]] = l_foox
+        axis_results[axis_idx["L FOOY"]] = l_fooy
+        axis_results[axis_idx["L FOOZ"]] = l_fooz
+
+        # Populate RFoot and LFoot in joint_centers
+        joint_centers[jc_idx['RFoot']] = r_fooo
+        joint_centers[jc_idx['LFoot']] = l_fooo
+
+        rfhd = frame[marker_idx[markers["RFHD"]]]
+        lfhd = frame[marker_idx[markers["LFHD"]]]
+        rbhd = frame[marker_idx[markers["RBHD"]]]
+        lbhd = frame[marker_idx[markers["LBHD"]]]
+
+        # Populate Front_Head and Back_Head in joint_centers
+        joint_centers[jc_idx['Front_Head']] = np.array([(rfhd + lfhd) / 2.0])
```

Comment:
This calc function should should be doing the sequence of function calls, not also calculating new data